rm-datasource/src/main/java/org/apache/seata/rm/datasource/undo/AbstractUndoLogManager.java

```diff
@@ -151,18 +161,28 @@ public void batchDeleteUndoLog(Set<String> xids, Set<Long> branchIds, Connection
         int xidSize = xids.size();
         int branchIdSize = branchIds.size();
         String batchDeleteSql = toBatchDeleteUndoLogSql(xidSize, branchIdSize);
-        try (PreparedStatement deletePST = conn.prepareStatement(batchDeleteSql)) {
+        String batchDeleteSubSql = toBatchDeleteSubUndoLogSql(xidSize, branchIdSize);
+        try (PreparedStatement deletePST = conn.prepareStatement(batchDeleteSql);
+             PreparedStatement deleteSubPST = conn.prepareStatement(batchDeleteSubSql)) {
```

Comment:
为什么不在上面的batch delete sql中直接增加删除sub的条件？
Why not add the condition to delete sub directly in the batch delete sql above?

Reply:
主log和子log的条件不一样

The conditions for the main log and the sub-log are different.

Reply:
> 主log和子log的条件不一样
> 
> The conditions for the main log and the sub-log are different.

为什么我们不能将其合并成一个sql？虽然条件不一样，可以是or的关系不是吗？
Why can't we merge it into one sql? Although the conditions are different, it can be an or relationship, right?

Reply:
一个是走索引，一个不走索引的
One is indexed, and the other is not indexed.

Reply:
ok